### DISCUSSION TITLE
spark.components.WindowedApplication and NativeDragEvent

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#2954] [Q&A]
Hello,

how to use [NativeDragEvent](https://help.adobe.com/fr_FR/FlashPlatform/reference/actionscript/3/flash/events/NativeDragEvent.html) (drag a file on a component) in `spark.components.WindowedApplication` ?

Thanks

**Comment by pol2095** [comment]
I solved it but I have the icon of file (too big on Windows) above my app, is-it possible to change this icon ?

**Reply by 2jfw** [reply]
How did you create the "Drag"? Like from inside your application by code (-> sample?)?..
If from outside, wouldn't it be Windows (/OS) handling the size of dragged items?

Anyways, if using the `DragManager`, did you try to add a `dragImage`?

Something like:

```
var scale : Number = 1;
var dragImage : Image = new Image();
var bitmapData : BitmapData   = new BitmapData(32 * scale,
			                                       32 * scale,
			                                       false,
			                                       0x000000);

dragImage.source = bitmapData;

DragManager.doDrag(this,
                    dragSource,
                    event,
                    dragImage,  [...]
``` 

EDIT: You can of course draw something (like an icon) into the bitmapData using `draw`.

> dragImage:[IFlexDisplayObject](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/mx/core/IFlexDisplayObject.html) (default = null) — The image to drag. This argument is optional. If omitted, a standard drag rectangle is used during the drag and drop operation. If you specify an image, you must explicitly set a height and width of the image or else it will not appear.

https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/mx/managers/DragManager.html#doDrag()

**Reply by pol2095** [reply]
The icon is generated by "Windows Explorer", when the mouse is hover "MS Word" for example the icon change, but not hover Google Chrome.
I don't use [doDrag()](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/mx/managers/DragManager.html#doDrag()) (for drag out gesture) but [acceptDragDrop()](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/mx/managers/DragManager.html#acceptDragDrop())  (for drag in gesture).

It's a Windows 11 stange behavior, when I drag an file on the Desktop, the icon is transparent (good behavior) but using "Windows Explorer", the icon is not transparent  (bad behavior).

**Reply by pol2095** [reply]
I asked [MS community](https://answers.microsoft.com/en-us/windows/forum/windows_11-files/windows-explorer-file-drag-icon-and-transparency/89694c15-fae7-49f3-bb40-aee77ac7cece), it's a Windows 11 bug.
